### PR TITLE
MODINVSTOR-624: Upgrade to Vert.x 4 (WIP)

### PR DESCRIPTION
**Pull request by julianladisch**
See RMB upgrade to Vert.x 4: https://github.com/folio-org/raml-module-builder/pull/789

This is not intended to get built on Jenkins!

To test this pull request you need to build the RMB pull request above as 32.0.0-SNAPSHOT on your local machine and then build this mod-inventory-storage pull request branch.

This pull request is a temporary branch intended for testing the RMB upgrade.

### CODE REVIEW

File: src/main/java/org/folio/rest/impl/InstanceStorageBatchAPI.java
Comment (adamdickmeiss) [inline src/main/java/org/folio/rest/impl/InstanceStorageBatchAPI.java:116]:
```suggestion
        .onComplete(ar -> saveFutures.addAll(futures))
        .map(always -> saveFutures);
```

**Comment by julianladisch** [comment]
I approve this pull request, it builds when using https://github.com/folio-org/raml-module-builder/pull/789

**Review by MikhailFokanov — APPROVED** [review]
looks good

**Review by hjiebsco — APPROVED** [review]

**Comment by julianladisch** [comment]
I approve this pull request.

**Comment by adamdickmeiss** [comment]
Thx @hjiebsco .. But we need this approved first.. https://github.com/folio-org/okapi/pull/1053 . Otherwise mod-inventory-storage will not be able to be enabled for a tenant anywhere.

**Comment by marcjohnson-kint** [comment]
> Thx @hjiebsco .. But we need this approved first.. [folio-org/okapi#1053](https://github.com/folio-org/okapi/pull/1053) . Otherwise mod-inventory-storage will not be able to be enabled for a tenant anywhere.

Is that because the upgrade to vert.x 4.0 (and a new version of RAML Module Builder) also changes the version of the Tenant API provided?

**Comment by marcjohnson-kint** [comment]
@MikhailFokanov I thought @adamdickmeiss just suggested that this *should not* be merged until the Okapi work it is blocked on was completed, why has this been merged?

**Comment by MikhailFokanov** [comment]
We dissussed with @julianladisch and @skoczko that it should be merged on the meeting 30 mins ago. I waited for approval from @julianladisch. When I see it in email notifiaction, I merged it. I am sorry, that I missed the new comments by  @adamdickmeiss and @marcjohnson-kint (posted less than 20 mins ago).

**Comment by marcjohnson-kint** [comment]
@MikhailFokanov 

> I am sorry, that I missed the new comments by @adamdickmeiss and @marcjohnson-kint 

That's ok, I wasn't blaming you, only trying to understand what happened to cause something to be merged when it shouldn't have been.

**Pull request merged**
